Sources/StreamChatUI/Composer/ComposerVC.swift

```diff
@@ -0,0 +1,645 @@
+            )
+        }
+    }
+
+    /// The content of the composer.
+    public var content: Content = .empty() {
+        didSet {
+            updateContentIfNeeded()
+        }
+    }
+
+    /// The delegate of the ComposerVC that notifies composer events.
+    open weak var delegate: ComposerVCDelegate?
+
+    /// A symbol that is used to recognise when the user is mentioning a user.
+    open var mentionSymbol = "@"
+
+    /// A symbol that is used to recognise when the user is typing a command.
+    open var commandSymbol = "/"
+
+    /// A controller to search users and that is used to populate the mention suggestions.
+    open var userSearchController: _ChatUserSearchController<ExtraData>!
+
+    /// A controller that manages the channel that the composer is creating content for.
+    open var channelController: _ChatChannelController<ExtraData>!
+
+    /// The view of the composer.
+    open private(set) lazy var composerView: _ComposerView<ExtraData> = components
+        .messageComposer
+        .messageComposerView.init()
+        .withoutAutoresizingMaskConstraints
+
+    /// The view controller that shows the suggestions when the user is typing.
+    open private(set) lazy var suggestionsVC: _ChatSuggestionsViewController<ExtraData> = components
+        .messageComposer
+        .suggestionsViewController
+        .init()
+
+    /// The view controller for selecting image attachments.
+    open private(set) lazy var imagePickerVC: UIViewController = {
+        let picker = UIImagePickerController()
+        picker.mediaTypes = ["public.image"]
+        picker.sourceType = .photoLibrary
+        picker.delegate = self
+        return picker
+    }()
+
+    /// The view controller for selecting file attachments.
+    open private(set) lazy var filePickerVC: UIViewController = {
+        let picker = UIDocumentPickerViewController(documentTypes: ["public.item"], in: .import)
+        picker.delegate = self
+        picker.allowsMultipleSelection = true
+        return picker
+    }()
+
+    public enum SelectedAttachments {
+        case media, documents
+    }
+
+    open var selectedAttachments: SelectedAttachments? {
+        if content.imageAttachments.isEmpty, content.documentAttachments.isEmpty {
+            return .none
+        } else {
+            return content.imageAttachments.isEmpty ? .documents : .media
+        }
+    }
+
+    open var attachments: [AttachmentEnvelope] {
+        switch selectedAttachments {
+        case .media:
+            return content.imageAttachments
+                .map(\.localURL)
+                .compactMap(AttachmentEnvelope.init)
+        case .documents:
+            return content.documentAttachments
+                .map(\.localURL)
+                .compactMap(AttachmentEnvelope.init)
+        case .none:
+            return []
+        }
+    }
+
+    public func setDelegate(_ delegate: ComposerVCDelegate) {
+        self.delegate = delegate
+    }
```

Comment:
What are the rules to use the `setDelegate` function or just set `var delegate` from outside?

Reply:
There are no rules yet I think, this function was just to keep consistency with LLC 🤔 But I guess we should opt for one way or another, in the whole codebase. CC @VojtaStavik

Reply:
Yes, please 🙌🏻